### ISSUE TITLE
ngDialog does not disappear on Chrome browser

### ISSUE THREAD

**Issue by ashis1973** [likeastore/ngDialog#199]
Hello,

I have an angularjs app where I am using ngdialog to let user choose some value based on a condition. It appears the dialog is doing what it intends to do. Its firing the closing even as well but the only problem is the dialog itself is not going away when I cam closing it.

I am using angularjs routing for the app.

Any help is highly appreciated.

//works fine
$scope.openGroupModal = function(user){
            $log.log("modal window is about to popup");
            $scope.user = user;
            $scope.dialog = ngDialog.open({ template: 'app/views/groupModal.html',className: 'ngdialog-theme-plain',
            scope: $scope
                /*,
            closeByEscape: false,
                preCloseCallback: function(value) {
                    if(confirm('Are you sure you want to close without saving your changes?')) {
                        return true;
                    }
                    return false;
                }*/
        });
        };
//works fine as well
        $scope.closeGroupModal = function(result){
            $log.log("closing dialog");
            //ngDialog.close($scope.dialog.id);
            ngDialog.close(result);
            $location.path('/manageHesUsers');
        };

**Comment by PavelPolyakov** [comment]
Same for me!

Suddenly, closing has stopped to work.

My research ended up with this piece of code:
~~~javascript
$dialog.unbind(animationEndEvent).bind(animationEndEvent, function () {
								$dialog.remove();
								if (dialogsCount === 0) {
									$body.removeClass('ngdialog-open');
									privateMethods.resetBodyPadding();
								}
								$rootScope.$broadcast('ngDialog.closed', $dialog);
							}).addClass('ngdialog-closing');
~~~
( https://github.com/likeastore/ngDialog/blob/master/js/ngDialog.js#L147 )

As far as animation is enabled - I am not able to close the dialog in Chrome. The callback is never called.

**Comment by PavelPolyakov** [comment]
Any idea which animation could conflict with the ngDialog?

**Comment by ashis1973** [comment]
Thanks for sharing your idea Pavel!

Here is my issue if I try to execute this code :+1:  $scope.dialog.unbind(animationEndEvent).bind(animationEndEvent,function(){
            $log.log('removing dialog');
            $scope.dialog.remove();
        });
Cannot read property 'unbind' of null.

$scope.dialog is the place holder of ngDialog.open(); and it is not null in this case.

Any idea what I am doing wrong here?

Thanks!

**Comment by PavelPolyakov** [comment]
Well you just should not execute this code from your code.
It's either work either not.

And I don't know why it doesn't work as it supposed to.

btw, when pasting the code samples, you may want to use the [markdown](https://guides.github.com/features/mastering-markdown/), so it would look nice.

**Comment by ashis1973** [comment]
Thanks! I will keep that in mind.

**Comment by ashis1973** [comment]
if I comment this line in ngDialog.js, the popup gets removed after hitting the close button.
//$dialog.unbind(animationEndEvent).bind(animationEndEvent, function () {

I do not see any issue so far after running my application having commented this line. 
Does anybody see any issue with this line being commented?

**Comment by PavelPolyakov** [comment]
As hardcode solution you can do the next. On the line 26, assign the ```false``` to the var ```animationEndSupport```.
https://github.com/likeastore/ngDialog/blob/master/js/ngDialog.js#L26

It would do the trick.

But, I expect that the code from the repository should work like it's designed. I don't know what had caused the fact that it stopped to work, it could be the Chrome update, for example.

Waiting for the contributors response :)

**Comment by amol-patil** [comment]
I am also having this problem in Chrome, for the simplest use case.

**Comment by PavelPolyakov** [comment]
@voronianski 

Please, have a look at the issue. It seems that it takes place in Chrome.

Regards,

**Comment by amol-patil** [comment]
So we just tested this on FF, Chrome and Safari and the problem occurs in all of these.

**Comment by thoughtpalette** [comment]
+1 Have the same issue in Safari/Mac Yosemite. 

Setting `animationEndSupport: false` works

**Comment by egor-smirnov** [comment]
Do you have this problem when you run `example/index.html`? I just checked them in Chrome / FF / Safari on Mac - everything seems fine.

If `example/index.html` works for you, then probably this is a problem with your particular case. Then jsfiddle / jsbin will be needed from your side.

**Comment by cezarykluczynski** [comment]
I had this problem when i decided to remove animation from dialog completely, after which dialog could not be closed.

My fix (SASS style) for default template, that works, at least in Chrome:

```sass
.ngdialog.ngdialog-theme-default.ngdialog-closing .ngdialog-content,
.ngdialog.ngdialog-theme-default .ngdialog-content,
.ngdialog.ngdialog-theme-default .ngdialog-button:focus,
.ngdialog-overlay,
.ngdialog.ngdialog-closing .ngdialog-overlay,
.ngdialog-content,
.ngdialog.ngdialog-closing .ngdialog-content
    -webkit-animation: none !important
    animation: none !important

.ngdialog.ngdialog-closing .ngdialog-overlay,
.ngdialog.ngdialog-closing .ngdialog-content
    -webkit-animation: ngdialog-flyin .0s !important
    animation: ngdialog-flyin .0s !important
```

Despite that, relying on animationend event to remove a dialog from DOM is not a good idea. Someone could decide to remove animations completely, or decide to have some long-running animation on the element, not related to the dialog closing effect. Espacially in that second case, dialog will never close.

**Comment by hancock808** [comment]
I replaced the animationEndEvent with a timeout set to the duration of the animation.

    if (animationEndSupport && !options.disableAnimation) {
        scope.$destroy();
        $timeout(function () {
            privateMethods.closeDialogElement($dialog, value);
        }, 500);
        //$dialog.unbind(animationEndEvent).bind(animationEndEvent, function () {
        //    privateMethods.closeDialogElement($dialog, value);
        //});
        $dialog.addClass('ngdialog-closing');
    }

**Comment by faceleg** [comment]
Does this still occur with 0.5.4?

**Comment by Klemen1337** [comment]
@faceleg yes it still occurs. For me ngdialog-content closes, but the ngdialog div is not removed. User then cannot interact with the website because of the ngdialog overlay.
@cezarykluczynski removing the animation worked, but I don't want to loose the fadeIn animation...

**Comment by cs-NET** [comment]
I have implemented @hancock808 solution and it seems like a good workaround. 

For me it looks like the animation end event will eventually get called, it just takes from 1 to 6 seconds during which the user can not interact with the website as the dialog overlay is still there.

**Comment by timroes** [comment]
I have the same behavior as @cs-NET. It takes around 1 - 5 seconds when closing a dialog until the page becomes clickable and scrollable again. Very annoying. You can see this in action here: http://www.devfestka.de/programm Just click some of the slots and close the dialog. It might not work directly, but reliable starts for me after some dialogs. Also problematic: as long as the dev tools are open in chrome, the behavior doesn't seem to occur.

**Comment by cs-NET** [comment]
I did a bit more digging around and it appears that it is a common problem with the animationEndEvent not firing reliably. I also wanted to add that the timeout workaround can cause a flashing of the dialog if the timeout fires after the animation. To combat this I set the timeout to 100 ms shorter than the animation.

**Comment by faceleg** [comment]
Can has PR?

**Comment by fultonm** [comment]
Having this issue as well. Seems like basic feature of such a library.
